detect.py

```diff
@@ -0,0 +1,374 @@
+            notes = contents[2] 
+
+            # append to ubuntu_packages_list 
+            ubuntu_packages_list.append(ubuntu_package)
+
+            # add to ubuntu_packages_dict 
+            ubuntu_packages_dict[ubuntu_package] = [mariner_replacement, notes]
+
+    # COMMANDS: open commands.csv and read into data structures
+    with open(commands) as csvfile:
+        # use csv.reader to parse
+        data = csv.reader(csvfile, delimiter = ',')
+        
+        # loop through row by row
+        for contents in data:
+            # set each element in list properly 
+            ubuntu_command = contents[0]
+            mariner_replacement = contents[1]
+            notes = contents[2] 
+
+            # append to ubuntu_commands_list 
+            ubuntu_commands_list.append(ubuntu_command)
+
+            # add to ubuntu_commands_dict 
+            ubuntu_commands_dict[ubuntu_command] = [mariner_replacement, notes]
+    
+    # FILEPATHS: open filepaths.csv and read into data structures
+    with open(filepaths) as csvfile:
+        # use csv.reader to parse
+        data = csv.reader(csvfile, delimiter = ',')
+   
+        # loop through row by row 
+        for contents in data:
+            # set each element in list properly 
+            ubuntu_fp = contents[0]
+            mariner_replacement = contents[1]
+            notes = contents[2] 
+
+            # append to ubuntu_filepaths_list 
+            ubuntu_filepaths_list.append(ubuntu_fp)
+
+            # add to ubuntu_filepaths_dict 
+            ubuntu_filepaths_dict[ubuntu_fp] = [mariner_replacement, notes]
+
+    # return to main
+    return True
+
+
+def main():
+    # if singular file or directory is inputted (argv = 2), determine if file or directory, and call proper file 
+    if len(sys.argv) == 2:
+        # put path in proper string format 
+        path = str(sys.argv[1])
+        
+        # check if path is usage
+        if path == "usage":
+            usage()
+        
+        # call csv_reader on data files 
+        csv_reader("database/packages.csv", "database/commands.csv", "database/filepaths.csv")
+        
+        # if path is a file call file_scan
+        if os.path.isfile(path):
+            file_scan(path)
+        
+        # elif path is a directory call dir_scan
+        elif os.path.isdir(path):
+            dir_scan(path)
+        
+        # if it is niether print error and call usage 
+        else:
+            print("ERROR: Inputted invald file/directory path.")
+            print("")
+            usage()
+   
+    # call usage if argv != 2 
+    else:
+        usage()
+
+    # call output_csv to output data file for both packages and commands 
+    output_csv1(ubuntu_packages_dict, found_packages, "package_data")
+    output_csv1(ubuntu_commands_dict, found_commands, "command_data")
+    output_csv2(ubuntu_filepaths_dict, found_filepaths, "filepath_data")
+
+    return True 
+
+
+if __name__ == '__main__':
```

Comment:
Consider removing some comments, where the code is obvious. You can comment if code has logic that's not obvious, otherwise having a general comment describing what the function is doing should suffice. Also consider removing extra blank lines.
For example, you can remove comments like 'call main', 'check if path is usage', 'append to ubuntu_filepaths_list', 'append to ubuntu_filepaths_list', 'get line and strip', etc.